src/http/http_request.f90

```diff
@@ -9,17 +14,31 @@ module http_request
     public :: request_type
 
     ! HTTP methods:
+    ! An integer parameter representing the HTTP GET method.
     integer, parameter :: HTTP_GET = 1
+    ! An integer parameter representing the HTTP HEAD method.
     integer, parameter :: HTTP_HEAD = 2
+    ! An integer parameter representing the HTTP POST method.
     integer, parameter :: HTTP_POST = 3
+    ! An integer parameter representing the HTTP PUT method.
     integer, parameter :: HTTP_PUT = 4
+    ! An integer parameter representing the HTTP DELETE method.
     integer, parameter :: HTTP_DELETE = 5
+    ! An integer parameter representing the HTTP PATCH method.
     integer, parameter :: HTTP_PATCH = 6
 
     ! Request Type
     type :: request_type
-        character(len=:), allocatable :: url, data
+    !! A derived type representing an HTTP request.
+        character(len=:), allocatable :: url, data, form_encoded_str
+            !!  url: a character allocatable component representing the URL of the request.
+            !! data: a character allocatable component representing the request data.
+            !! form_encoded_str: a character allocatable component representing the URL-encoded form data.
```

Comment:
Here, and elsewhere, you need to break the arguments each into their own line and have one line of documentation per argument. (see how I did it for the `form_type` above).